### PR TITLE
EVA-3585 Cli documentation

### PR DESCRIPTION
**Pull request by Dona094**
Eva-sub-cli documentation review

### CODE REVIEW

File: Getting_Started_with_eva_sub_cli.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
+
+```
+
+Body: The body of the VCF file contains the actual variant data, with each row representing a single variant. The columns in the body are : CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO, FORMAT, Sample Columns
+
+```
+#CHROM  POS  ID  REF  ALT  QUAL  FILTER  INFO  FORMAT  [SampleIDs...]
+``` 
+Here's a small example to illustrate the structure of a VCF file: Example VCF file archived at EVA to be inserted
```
Comment (nitin-ebi) [inline Getting_Started_with_eva_sub_cli.md:31]: **Example VCF file archived at EVA to be inserted**  is this a TODO ?

**Review by nitin-ebi — APPROVED** [review]

File: Getting_Started_with_eva_sub_cli.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- One or several valid VCF files
+- Completed metadata spreadsheet
+- Reference genome in fasta format
+
+The VCF file must adhere to official VCF specifications, and the metadata spreadsheet provides contextual information about the dataset. In the following sections, we will examine each of these inputs in detail.
+
+# VCF File
+
+A VCF (Variant Call Format) file is a type of file used in bioinformatics to store information about genetic variants. It includes data about the differences (or variants) between a sample's DNA and a reference genome. Typically, generating a VCF file involves several steps: preparing your sample, sequencing the DNA, aligning it to a reference genome, identifying variants, and finally, formatting this information into a VCF file. The overall goal is to systematically capture and record genetic differences in a standardised format. A VCF file consists of two main parts: the header and the body.
+Header: The header contains metadata about the file, such as the format version, reference genome information, and descriptions of the data fields. Each line in the header starts with a double ##, except for the last header line which starts with a single #.
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:14]:
```suggestion

Header: The header contains metadata about the file, such as the format version, reference genome information, and descriptions of the data fields. Each line in the header starts with a double ##, except for the last header line which starts with a single #.
```
Just to create a new paragraph in markdown.

File: Getting_Started_with_eva_sub_cli.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+File format version
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:16]: Guessing this is just a reminder for yourself? But somewhere in this section we could link to the [VCF specification](http://samtools.github.io/hts-specs/VCFv4.3.pdf).

File: Getting_Started_with_eva_sub_cli.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+| Project           | The objective of this sheet is to gather general information about the Project including submitter, submitting centre, collaborators, project title, description and publications. |
+| Sample            | Projects consist of analyses that are run on samples. We accept sample information in the form of BioSample, ENA or EGA accession(s). We also accept BioSamples sampleset accessions. If your samples are not yet accessioned, and are therefore novel, please use the "Novel sample(s)" sections of the Sample(s) worksheet to have them registered at BioSample |
+| Analysis          | For EVA, each analysis is one vcf file, plus an unlimited number of ancillary files. This sheet allows EVA to link vcf files to a project and to other EVA analyses. Additionally, this worksheet contains experimental meta-data detailing the methodology of each analysis. Important to note; one project can have multiple associated analyses          |
+| Files             | Filenames and associated checking data associated with this EVA submission should be entered into this worksheet. Each file should be linked to one, or more, analysis.            |
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:44]:
```suggestion
| Files             | Filenames and associated checking data associated with this EVA submission should be entered into this worksheet. Each file should be linked to exactly one analysis.            |
```
Not sure if we're thinking of changing this in the future, but I believe this is what's supported now.

File: Getting_Started_with_eva_sub_cli.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+The CLI tool performs the following validation checks and generates corresponding reports:
+
+- Metadata check to ensure that the metadata fields have been correctly filled in
+- VCF check to ensure that the VCF file follows the VCF format specification
+- Assembly check to ensure that the genome and the VCF match
+- Sample name check to ensure that the samples in the metadata can be associated with the sames in the VCF
+
+In the following sections, we will examine each of these checks in detail, starting with the Metadata check.
+
+# Metadata check
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:58]:
```suggestion
## Metadata check
```
Optional, this makes a smaller header, so this becomes a subsection of "Validation checks"... would do this for all the other three headers as well.

File: Getting_Started_with_eva_sub_cli.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+# Sample Name Concordance Check
+
+The sample name concordance check ensures that the sample names in the metadata spreadsheet match those in the VCF file. This is achieved by cross-checking the 'Sample name in VCF' column in the spreadsheet with the sample names registered in the VCF file. Any discrepancies must be addressed by the submitter when the CLI tool generates a report of the  mismatches found.
+
+Key Points to note Before validating your data with the eva-sub-cli Tool:
+
+- Ensure that sample names between the VCF file and the metadata spreadsheet match. This comparison is case-sensitive.
+- Ensure there are no extra spaces in the sample names.
+
+Common Errors Seen with Sample Concordance Checks:
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:121]:
```suggestion
Common errors seen with sample concordance checks:
```

File: Getting_Started_with_eva_sub_cli.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+ 
+- VCF file uses a non-GCA contig alias causing the assembly check to fail
+- Contigs used do not exist in the assembly report of the reference genome
+- Major Allele Used as REF Allele: This typically occurs when a specific version of Plink or Tassel is used to create VCF files, causing the tool to use the major allele as the reference allele. In such cases, submitters should use the GCA FASTA sequence to create corrected files.
+
+# Sample Name Concordance Check
+
+The sample name concordance check ensures that the sample names in the metadata spreadsheet match those in the VCF file. This is achieved by cross-checking the 'Sample name in VCF' column in the spreadsheet with the sample names registered in the VCF file. Any discrepancies must be addressed by the submitter when the CLI tool generates a report of the  mismatches found.
+
+Key Points to note Before validating your data with the eva-sub-cli Tool:
```
Comment (apriltuesday) [inline Getting_Started_with_eva_sub_cli.md:116]:
```suggestion
Key points to note before validating your data with the eva-sub-cli tool:
```

File: Getting_Started_with_eva_sub_cli.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+
+# Validation checks 
+
+The CLI tool performs the following validation checks and generates corresponding reports:
+
+- Metadata check to ensure that the metadata fields have been correctly filled in
+- VCF check to ensure that the VCF file follows the VCF format specification
+- Assembly check to ensure that the genome and the VCF match
+- Sample name check to ensure that the samples in the metadata can be associated with the sames in the VCF
```
[the remark this answers is not in the document]
Comment (apriltuesday) [reply Getting_Started_with_eva_sub_cli.md:54]: I think this should actually be "samples"

**Review by apriltuesday — APPROVED** [review]
This is great! In particular, the sections on validation key points/common errors are something I think we've sorely needed for a while, and are very clearly written.

File: Getting_Started_with_eva_sub_cli.md (hunk 1 above)
Comment (Dona094) [reply Getting_Started_with_eva_sub_cli.md:31]: @nitin-ebi yes still searching for a good example vcf file that we have archived at EVA

File: Getting_Started_with_eva_sub_cli.md (hunk 3 above)
Comment (Dona094) [reply Getting_Started_with_eva_sub_cli.md:16]: great idea @apriltuesday

**Review by tcezard — APPROVED** [review]

**Pull request merged**